### DISCUSSION TITLE
Multiple correct answers in pl-order-blocks?

### DISCUSSION THREAD

**Discussion by dowobeha** [PrairieLearn/PrairieLearn#10664] [General]
I'm wondering if there's a way to allow multiple correct answers in pl-order-blocks. In this case, I don't think topological order will work.

My use case is situations like this, where there are multiple correct answers:

```
<pl-order-blocks answers-name="empty-vector" format="code" code-language="c++" solution-placement="bottom" inline="true">
  <pl-answer correct="true">auto</pl-answer>
  <pl-answer correct="true">my_data</pl-answer>
  <pl-answer correct="true">=</pl-answer>
  <pl-answer correct="true">std::vector&lt;int&gt;</pl-answer>
  <pl-answer correct="true"></pl-answer>
  <pl-answer correct="true">{</pl-answer>
  <pl-answer correct="true">}</pl-answer>
  <pl-answer correct="true">;</pl-answer>
</pl-order-blocks>
```

The code above will accept `auto my_data = std::vector<int>{};`, but I'd like to also accept as correct `std::vector<int> my_data{};` and `std::vector<int> my_data = {};`

**Comment by nwalters512** [comment]
There's currently no out-of-the-box support for this. See https://github.com/PrairieLearn/PrairieLearn/issues/4668 for a discussion of this feature request. It's largely focused around writing proofs, but maybe it's generalizable to what you're asking for!

**Reply by dowobeha** [reply]
I would be fine with handing the grading in server.py, but it isn’t clear to me if that’s possible, either

**Reply by jonatanschroeder** [reply]
My suggestion: use `grading-method="external" file-name="signature.h"`. This will cause a file named `signature.h` to be created with the appropriate values. You can then use the grade function in server.py (see #8488) to read the signature.h file and check for valid options.

**Reply by dowobeha** [reply]
Thanks. Is there not a simpler approach to directly access the student answer in server.py without the need to write to and subsequently read a file?

**Reply by jonatanschroeder** [reply]
You could parse the result of `data["submitted_answers"]["empty-vector"]`, which I believe is an array of objects in the order provided in the submission, where the `inner_html` of each object contains the text of the answer.
